src/silo/config/util/yaml_file.cpp

```diff
@@ -0,0 +1,51 @@
+#include "silo/config/util/yaml_file.h"
+
+#include <fmt/format.h>
+#include <spdlog/spdlog.h>
+#include <boost/lexical_cast.hpp>
+
+using silo::config::YamlFile;
+
+YamlFile::YamlFile(const std::filesystem::path& filename)
+    : filename(filename) {
+   SPDLOG_INFO("Reading config from {}", filename.string());
+   try {
+      node = YAML::LoadFile(filename.string());
+   } catch (const YAML::Exception& e) {
+      throw std::runtime_error(
+         fmt::format("Failed to read preprocessing config from {}: {}", filename.string(), e.what())
+      );
+   }
+}
+
+std::string YamlFile::configType() const {
+   return fmt::format("yaml config file '{}'", filename.string());
+}
+
+bool YamlFile::hasProperty(const Option& option) const {
+   YAML::Node current = Clone(node);
+   for (const auto& access : option.access_path) {
+      if (!current.IsDefined() || !current.IsMap()) {
+         return false;
+      }
+      current = current[access];
+   }
+   if (!current.IsDefined() || !current.IsMap()) {
+      return false;
+   }
+   return true;
```

Comment:
```suggestion
   return current.IsDefined() && current.IsMap();
```

Reply:
I'm not even sure whether this is correct. `IsMap` seems strange here. I think a unit test would be good that covers this method and `getString`.